Ubuntu 20.04 and Fedora 32 build results

### Version of Beast

299
### Steps necessary to reproduce the problem

(from beast/pull/2028)

docker build -f .dockers/ubuntu-20/Dockerfile .

### Summary of result

```
...failed updating 2 targets...
...updated 4200 targets...

libs/beast/test/beast/core/basic_stream.cpp:1346:20: error: no matching member function for call to 'async_connect'
```
### Further output

```
clang-linux.compile.c++.without-pch bin.v2/libs/beast/test/beast/core/basic_stream.test/clang-linux-10.0.0/release/cxxstd-2a-iso/threadapi-pthread/threading-multi/visibility-hidden/basic_stream.o
In file included from libs/beast/test/beast/core/basic_stream.cpp:26:
In file included from ./boost/asio/spawn.hpp:19:
In file included from ./boost/coroutine/all.hpp:11:
In file included from ./boost/coroutine/coroutine.hpp:10:
In file included from ./boost/coroutine/asymmetric_coroutine.hpp:24:
In file included from ./boost/coroutine/exceptions.hpp:14:
 ./boost/detail/scoped_enum_emulation.hpp:17:1: warning: This header is deprecated. Use <boost/core/scoped_enum.hpp> instead. [-W#pragma-messages]
BOOST_HEADER_DEPRECATED("<boost/core/scoped_enum.hpp>")
^
./boost/config/header_deprecated.hpp:23:37: note: expanded from macro 'BOOST_HEADER_DEPRECATED'
# define BOOST_HEADER_DEPRECATED(a) BOOST_PRAGMA_MESSAGE("This header is deprecated. Use " a " instead.")
                                  ^
./boost/config/pragma_message.hpp:24:34: note: expanded from macro 'BOOST_PRAGMA_MESSAGE'
# define BOOST_PRAGMA_MESSAGE(x) _Pragma(BOOST_STRINGIZE(message(x)))
                             ^
 <scratch space>:133:2: note: expanded from here
message("This header is deprecated. Use " "<boost/core/scoped_enum.hpp>" " instead.")
^
libs/beast/test/beast/core/basic_stream.cpp:1346:20: error: no matching member function for call to 'async_connect'
             stream.async_connect(
             ~~~~~~~^~~~~~~~~~~~~
./boost/beast/core/basic_stream.hpp:1203:5: note: candidate template ignored: constraints not satisfied [with Iterator = boost::asio::ip::basic_resolver_iterat     or<boost::asio::ip::tcp>, ConnectCondition = condition, IteratorConnectHandler = handler]
    async_connect(
    ^
./boost/beast/core/basic_stream.hpp:1196:9: note: because '::boost::asio::completion_token_for<handler, void (boost::beast::error_code, boost::asio::ip::basic_     resolver_iterator<boost::asio::ip::tcp>)>' evaluated to false
         BOOST_ASIO_COMPLETION_TOKEN_FOR(
       ^
./boost/asio/async_result.hpp:502:18: note: expanded from macro 'BOOST_ASIO_COMPLETION_TOKEN_FOR'
::boost::asio::completion_token_for<s>

--- a/.dockers/ubuntu-20/user-config.jam
+++ b/.dockers/ubuntu-20/user-config.jam
@@ -7,5 +7,5 @@ feature.compose <asio.mode>ts : <define>"BOOST_ASIO_USE_TS_EXECUTOR_AS_DEFAULT" 
 feature.compose <asio.mode>nodep-nots : <define>"BOOST_ASIO_NO_DEPRECATED" <define>"BOOST_ASIO_NO_TS_EXECUTORS" ;
 feature.compose <asio.mode>nodep-ts : <define>"BOOST_ASIO_NO_DEPRECATED" <define>"BOOST_ASIO_USE_TS_EXECUTOR_AS_DEFAULT" ;
 
-using clang :   : clang++ : <stdlib>"libc++" <cxxflags>"-Wno-c99-extensions" ;
+using clang :   : clang++ : <cxxflags>-stdlib=libc++ <cxxflags>"-DBOOST_ASIO_DISABLE_CONCEPTS" <cxxflags>"-Wno-c99-extensions" <linkflags>-stdlib=libc++ ;
 using gcc :   : g++ : <cxxflags>"-Wno-c99-extensions" ;

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,6 @@
+* Add Bishop Fox Security Assessment.
+--------------------------------------------------------------------------------
+
 Version302:
 
 * Fix assert when basic_stream used as underlying of ssl::stream with zero-length write.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,7 +1,9 @@
+Version 303:
+
 * Add Bishop Fox Security Assessment.
 --------------------------------------------------------------------------------
 
-Version302:
+Version 302:
 
 * Fix assert when basic_stream used as underlying of ssl::stream with zero-length write.
 * Add Sec-* HTTP headers.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -40,7 +40,7 @@ endfunction()
 #
 #-------------------------------------------------------------------------------
 
-project (Beast VERSION 302)
+project (Beast VERSION 303)
 
 set_property (GLOBAL PROPERTY USE_FOLDERS ON)
 option (Beast_BUILD_EXAMPLES "Build examples" ON)

--- a/doc/qbk/01_intro/1a_bishop_fox.qbk
+++ b/doc/qbk/01_intro/1a_bishop_fox.qbk
@@ -30,6 +30,14 @@ manual exploitation and review of these issues to validate the findings.
 
 [@https://vinniefalco.github.io/BeastAssets/Beast%20-%20Hybrid%20Application%20Assessment%202017%20-%20Assessment%20Report%20-%2020171114.pdf [*Beast - Hybrid Application Assessment 2017]]
 
+In 2020, as part of its commitment to producing the very finest C++ libraries
+that application developers can trust, the C++ Alliance one again comissioned
+Bishop Fox to retest the Beast library.
+
+The report is linked here:
+
+[@https://vinniefalco.github.io/BeastAssets/C%20Plus%20Plus%20Alliance%20-%20Hybrid%20Application%20Assessment%202020%20-%20Assessment%20Report%20-%2020200924%20.pdf [*Beast - Hybrid Application Assessment 2020]]
+
 [/ "Securing Boost.Beast: A Non-Traditional Source Code Review"]
 '''
 <mediaobject>

--- a/doc/qbk/01_intro/1a_bishop_fox.qbk
+++ b/doc/qbk/01_intro/1a_bishop_fox.qbk
@@ -9,6 +9,15 @@
 
 [section:security_review_bishop_fox Security Review (Bishop Fox) __video__]
 
+In 2020, as part of its commitment to producing the very finest C++ libraries
+that application developers can trust, the C++ Alliance one again commissioned
+Bishop Fox to retest the Beast library.
+
+The report is linked here:
+
+[@https://github.com/CPPAlliance/beast-assets/raw/master/bishop-fox/C%20Plus%20Plus%20Alliance%20-%20Hybrid%20Application%20Assessment%202020%20-%20Assessment%20Report%20-%2020200924.pdf
+    [*Beast - Hybrid Application Assessment 2020]]
+
 Since 2005, [@https://www.bishopfox.com/ Bishop Fox] has provided
 security consulting services to the Fortune 1000, high-tech startups,
 and financial institutions worldwide.
@@ -28,15 +37,8 @@ addition, the team performs a review of the application architecture and
 business logic to locate any design-level issues. Finally, the team performs
 manual exploitation and review of these issues to validate the findings.
 
-[@https://vinniefalco.github.io/BeastAssets/Beast%20-%20Hybrid%20Application%20Assessment%202017%20-%20Assessment%20Report%20-%2020171114.pdf [*Beast - Hybrid Application Assessment 2017]]
-
-In 2020, as part of its commitment to producing the very finest C++ libraries
-that application developers can trust, the C++ Alliance one again comissioned
-Bishop Fox to retest the Beast library.
-
-The report is linked here:
-
-[@https://vinniefalco.github.io/BeastAssets/C%20Plus%20Plus%20Alliance%20-%20Hybrid%20Application%20Assessment%202020%20-%20Assessment%20Report%20-%2020200924%20.pdf [*Beast - Hybrid Application Assessment 2020]]
+[@https://github.com/madmongo1/beast-assets/raw/master/bishop-fox/Beast%20-%20Hybrid%20Application%20Assessment%202017%20-%20Assessment%20Report%20-%2020171114.pdf
+    [*Beast - Hybrid Application Assessment 2017]]
 
 [/ "Securing Boost.Beast: A Non-Traditional Source Code Review"]
 '''

--- a/include/boost/beast/core/detail/tuple.hpp
+++ b/include/boost/beast/core/detail/tuple.hpp
@@ -1,5 +1,5 @@
 //
-// Copyright (c) 2016-2019Damian Jarek (damian dot jarek93 at gmail dot com)
+// Copyright (c) 2016-2019 Damian Jarek (damian dot jarek93 at gmail dot com)
 //
 // Distributed under the Boost Software License, Version 1.0. (See accompanying
 // file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)

--- a/include/boost/beast/version.hpp
+++ b/include/boost/beast/version.hpp
@@ -20,7 +20,7 @@
     This is a simple integer that is incremented by one every
     time a set of code changes is merged to the develop branch.
 */
-#define BOOST_BEAST_VERSION 302
+#define BOOST_BEAST_VERSION 303
 
 #define BOOST_BEAST_VERSION_STRING "Boost.Beast/" BOOST_STRINGIZE(BOOST_BEAST_VERSION)
 

--- a/test/beast/core/_detail_tuple.cpp
+++ b/test/beast/core/_detail_tuple.cpp
@@ -1,5 +1,5 @@
 //
-// Copyright (c) 2016-2019Damian Jarek (damian dot jarek93 at gmail dot com)
+// Copyright (c) 2016-2019 Damian Jarek (damian dot jarek93 at gmail dot com)
 //
 // Distributed under the Boost Software License, Version 1.0. (See accompanying
 // file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)

